### DISCUSSION TITLE
FYI: IE to finally retire June 2022

### DISCUSSION THREAD

**Discussion by twMat** [TiddlyWiki/TiddlyWiki5#5875] [General]
https://blogs.windows.com/windowsexperience/2021/05/19/the-future-of-internet-explorer-on-windows-10-is-in-microsoft-edge/

If not earlier, I guess that this should be the last date that we need to pay consideration to IE.

Maybe this thread or some other, could hold a collection of the things, or links to things, that we've been prevented from doing for IE considerations so they're not forgotten.

**Comment by pmario** [comment]
We could finally use some of the features up to ES9 aka ES2018 which are well supported by modern browsers. 
See: https://exploringjs.com/es6/ch_overviews.html and https://www.freecodecamp.org/news/here-are-examples-of-everything-new-in-ecmascript-2016-2017-and-2018-d52fa3b5a70e/

can I use ES9: https://caniuse.com/?feats=mdn-javascript_builtins_regexp_dotall,mdn-javascript_builtins_regexp_lookbehind_assertion,mdn-javascript_builtins_regexp_named_capture_groups,mdn-javascript_builtins_regexp_property_escapes,mdn-javascript_builtins_symbol_asynciterator,mdn-javascript_functions_method_definitions_async_generator_methods,mdn-javascript_grammar_template_literals_template_literal_revision,mdn-javascript_operators_destructuring_rest_in_objects,mdn-javascript_operators_spread_spread_in_destructuring,promise-finally

**Reply by pmario** [reply]
Using those specs should finally allow us to avoid the the hardcoded `tags.sort()` output. So we should be able to give tags an order.

**Reply by Jermolene** [reply]
Hi @pmario I don't quite understand; tags already have an ordering, what are you referring to?

**Reply by pmario** [reply]
If I do enter tags: "a" "c" "b" I do get "a" "b" "c" ... That's not what I want. I want it to be as I inserted it. 

The same should be true for the "fields list". I want to be able to sort the fields in a way I want and not alphabetically. ... 

May be tags have been sorted alphabetically because of consistency reasons. ... BUT imo fields have to be sorted, because we couldn't guarantee the sort order, due to browser implementation details.

**Reply by pmario** [reply]
Tags have an order because there is the "list" field. ... I want to be able to use every field and it's content in an ordered way.

**Reply by Jermolene** [reply]
I see what you mean. What's the connection with ES2018?

**Reply by pmario** [reply]
With ES2019 the browser compatibility goes down for several functions. .. But you are right. We could define it a bit differently. 

So we may only use functions, that have >= eg: 93% of browser support. ... Where we already loose 7% of the overall market.

**Reply by pmario** [reply]
Or we need to start polyfilling some functions if possible. But this would increase the size quite a bit.

**Reply by linonetwo** [reply]
> loose 7% of the overall market

I don't think so, those users may not using tiddlywiki... I think tiddlywiki is somehow targeting poweruser, and poweruser will use as latest of browser as possible.

And using electron or nw based app makes sure less powerful users can use tw in a non-technical way, while still uses the latest browser features. So in conclusion most of the users may use tw in latest browsers...

**Comment by linonetwo** [comment]
Supermemo is still using IE as built-in browser. This will be a good chance for tw to replace supermemo in the field of incremental reading. (with fishing plugin)
